lib/signs/utilities/predictions.ex

```diff
@@ -150,19 +152,26 @@ defmodule Signs.Utilities.Predictions do
     |> Enum.take(1)
   end
 
-  @spec stopped_train?(Predictions.Prediction.t()) :: boolean()
-  defp stopped_train?(%{
-         seconds_until_arrival: arrival_seconds,
-         seconds_until_departure: departure_seconds
-       })
-       when arrival_seconds >= Content.Utilities.max_time_seconds() or
-              departure_seconds >= Content.Utilities.max_time_seconds() do
-    false
+  defp approximate_time?(sec, certainty) do
+    sec && (sec > 60 * 60 || (sec > 20 * 60 && certainty == @reverse_prediction_certainty))
   end
 
-  defp stopped_train?(prediction) do
-    status = prediction.boarding_status
-    status && String.starts_with?(status, "Stopped") && status != "Stopped at station"
+  @spec stopped_train?(Predictions.Prediction.t()) :: boolean()
+  defp stopped_train?(%{
+         boarding_status: boarding_status,
+         seconds_until_arrival: seconds_until_arrival,
+         seconds_until_departure: seconds_until_departure,
+         arrival_certainty: arrival_certainty,
+         departure_certainty: departure_certainty
+       }) do
+    # Note: This performs a similar (but not identical) calculation to the one in the Message
+    # code for determining whether a prediction will show an approximate time. Ideally they
+    # should both call the same logic.
```

Comment:
Any reason to not go ahead and unify that logic now? I may be missing precisely how they're not identical. Could we do something like calling the logic in `Content.Message.Predictions.compute_minutes/2` but only take the second element in the result(s)?

Reply:
The complicating factor is that `compute_minutes` depends on selecting either the departure or the arrival time, which is currently embedded in `terminal` and `non_terminal`, and choosing between those functions requires looking up the source, e.g. with `get_source_by_stop_and_direction`. Pulling apart those layers would have added a lot of bulk to this PR.